pkg/sdk/controllerutil/controllerutil.go

```diff
@@ -56,23 +57,35 @@ func WaitForDeletion(ctx context.Context, cl client.Reader, o client.Object) err
 	}, ctx.Done())
 }
 
-func SupportsOwnerReference(restMapper meta.RESTMapper, owner, dependent client.Object) (bool, error) {
+func SupportsOwnerReference(restMapper meta.RESTMapper, owner, dependent runtime.Object, depNamespace string) (bool, error) {
```

Comment:
This change seems like a regression from the perspective of controller-runtime's prefered type (`client.Object`). The existing code using `client.Object` type enables us to extract both type and object metadata (so no need to use `meta.Accessor`).

Also, what's the purpose of the `depNamespace` parameter? Seems like we're always just passing the empty string.